jsEngine/settings/Settings.ts

```diff
@@ -20,18 +21,92 @@ export class JsEnginePluginSettingTab extends PluginSettingTab {
 	}
 
 	display(): void {
-		this.containerEl.empty();
+		const containerEl = this.containerEl;
+		const settings = this.plugin.settings;
+		containerEl.empty();
 
-		if (!this.plugin.settings) {
+		if (!settings) {
 			return;
 		}
 
-		// this.containerEl.createEl('p', { text: 'Currently Empty, but there will be stuff here later.' });
+		new Setting(containerEl)
+			.setName('JS snippets (loaded on startup)')
+			.setHeading()
+			.addExtraButton(el => {
+				el.setTooltip('Reload snippets')
+					.setIcon('refresh-cw')
+					.onClick(() => this.display());
+			})
+			.addExtraButton(el => {
+				el.setTooltip('Open snippets folder')
+					.setIcon('folder-open')
+					.onClick(async () => await this.openStartupScriptsDirectory());
+			});
 
-		new Setting(this.containerEl).setName('Startup scripts').addButton(button => {
-			button.setButtonText('Manage').onClick(() => {
-				new StartupScriptsModal(this.plugin).open();
+		new Setting(containerEl)
+			.setName('Custom JS Snippets Folder')
+			.setDesc('The folder to search for JavaScript files to load')
+			.addText(el => {
+				el.setPlaceholder(this.getDefaultStartupScriptsDirectory())
+					.setValue(settings.startupScriptsDirectory ?? '')
+					.onChange(async (val: string) => {
+						if (val) {
+							settings.startupScriptsDirectory = normalizePath(val);
+						} else {
+							settings.startupScriptsDirectory = undefined;
+						}
+						await this.plugin.saveSettings();
+					});
 			});
-		});
+
+		const startupScriptsDirectory = this.app.vault.getFolderByPath(this.getStartupScriptsDirectory());
+		let startupScripts: TFile[] = [];
+		if (startupScriptsDirectory != null) {
+			startupScripts = this.listJSfilesInDirectory(startupScriptsDirectory);
+		}
+		if (startupScriptsDirectory == null || startupScripts.length == 0) {
+			new Setting(containerEl).setName('No JS snippets found').setDesc(`JS snippets are stored in "vault/${this.getStartupScriptsDirectory()}"`);
+		} else {
+			for (const file of startupScripts) {
+				new Setting(containerEl)
+					.setName(file.basename)
+					.setDesc(`Apply JS snippet from "vault/${file.path}"`)
+					.addToggle(el => {
+						el.setValue(settings.startupScripts.contains(file.path)).onChange(val => this.toggleStartupScript(file, val));
+					});
+			}
+		}
+	}
+
+	getDefaultStartupScriptsDirectory(): string {
+		return normalizePath(''.concat(this.app.vault.configDir, '/scripts'));
+	}
+
+	getStartupScriptsDirectory(): string {
+		return this.plugin.settings.startupScriptsDirectory ?? this.getDefaultStartupScriptsDirectory();
+	}
+
+	async openStartupScriptsDirectory(): Promise<void> {
+		const vault = this.app.vault;
+		const directory = this.plugin.settings.startupScriptsDirectory ?? this.getDefaultStartupScriptsDirectory();
```

Comment:
Doesn't this line do the same thing as the method above? Why not use that method?

Reply:
This Methode is older than the refactor which seperated that logic into a function.l and I overlooked it.

Will fix!